django/test/testcases.py

```diff
@@ -1181,29 +1181,37 @@ def setUpClass(cls):
         super().setUpClass()
         if not cls._databases_support_transactions():
             return
-        cls.cls_atomics = cls._enter_atomics()
-
-        if cls.fixtures:
-            for db_name in cls._databases_names(include_mirrors=False):
-                try:
-                    call_command('loaddata', *cls.fixtures, **{'verbosity': 0, 'database': db_name})
-                except Exception:
-                    cls._rollback_atomics(cls.cls_atomics)
-                    cls._remove_databases_failures()
-                    raise
-        pre_attrs = cls.__dict__.copy()
+        # Disable the durability check to allow testing durable atomic blocks
+        # in a transaction for performance reasons.
+        transaction.Atomic.ensure_durability = False
```

Comment:
Slightly outside my practical experience, to be honest - but do we need to consider thread safety here? Would this cause problems if two concurrently running tests (in a multithreaded environment) are changing the same class attribute?

Reply:
Good question. I don't know the answer myself. If this is a problem, I suspect we can store the flag somewhere that is threadsafe.

Reply:
None of the `TestCase` are thread-safe AFAIK (e.g. think of `django.core.mail.outbox`, [`isolate_apps`](https://github.com/django/django/blob/b9cf764be62e77b4777b3a75ec256f6209a57671/django/test/utils.py#L825), ...).

I think it's fine to have it live on `Atomic` but I guess it could also live on `connections` objects like other atomic related flags and these happen to be thread safe.

Reply:
Haha, I think I started with it on `connections` (https://github.com/django/django/pull/13708#discussion_r528603433).

If django testing isn't thread-safe anyway, I lean towards leaving it as is.

Reply:
Yeah tests aren't thread safe at all.

I asked earlier to move it from connection to `Atomic` since we only need a global behaviour, not per-connection... I think it's also fine to live on `Atomic`. We could call it `_ensure_durability` to flag it as "private" and comment in `Atomic` that it's for tests only to discourage use in other situations.

Reply:
Ok great.

+1 to prefixing the attribute with an underscore.

Reply:
Done.